src/en/mayotune/src/eu/kanade/tachiyomi/extension/en/mayotune/MayoTune.kt

```diff
@@ -0,0 +1,124 @@
+package eu.kanade.tachiyomi.extension.en.mayotune
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.parseAs
+import okhttp3.Request
+import okhttp3.Response
+import rx.Observable
+
+class MayoTune() : HttpSource() {
+    override val name: String = "MayoTune"
+    override val baseUrl: String = "https://mayotune.xyz"
+    override val lang: String = "en"
+    override val versionId: Int = 1
+
+    private val source = SManga.create().apply {
+        title = "Mayonaka Heart Tune"
+        url = "/"
+        thumbnail_url = "$baseUrl/img/cover.jpg"
+        author = "Masakuni Igarashi"
+    }
+
+    // Popular
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        return Observable.just(MangasPage(listOf(source), false))
+    }
+
+    override fun popularMangaRequest(page: Int): Request = throw UnsupportedOperationException()
+    override fun popularMangaParse(response: Response): MangasPage =
+        throw UnsupportedOperationException()
+
+    // Latest
+    override val supportsLatest: Boolean = true
+    override fun fetchLatestUpdates(page: Int): Observable<MangasPage> {
+        return Observable.just(MangasPage(listOf(source), false))
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request = throw UnsupportedOperationException()
+    override fun latestUpdatesParse(response: Response): MangasPage =
+        throw UnsupportedOperationException()
+
+    // Search
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList,
+    ): Observable<MangasPage> {
+        val mangas = mutableListOf<SManga>()
+
+        if (source.title.lowercase().contains(query.lowercase()) ||
+            source.author?.lowercase()?.contains(query.lowercase()) == true
+        ) {
+            mangas.add(source)
+        }
+
+        return Observable.just(MangasPage(mangas, false))
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList) =
+        throw UnsupportedOperationException()
+
+    override fun searchMangaParse(response: Response): MangasPage =
+        throw UnsupportedOperationException()
+
+    // Get Override
+    override fun chapterListRequest(manga: SManga): Request {
```

Comment:
need to override `getChapterUrl` to provide webview url for chapter (webview button in reader)

Reply:
no the `getChapterUrl` method of `HttpSource`, it should return website url instead of api url, which you return in `chapterListRequest`

Reply:
> if you open the reader in app, there is a webview button, this method is for that

That is how I got there, and it is behaving as expected on a fresh installation/AVD. Is it still required?

https://github.com/user-attachments/assets/eecf2de4-0e21-4ca3-87be-67b56eddf411

Reply:
ah sorry, I thought that method uses the `chapterListRequest`, it actually uses `pageListRequest`...  mb